common/src/main/java/tachyon/util/io/FileUtils.java

```diff
@@ -121,20 +121,12 @@ public static void setLocalFileStickyBit(String file) {
    * @throws java.io.IOException
    */
   public static void createBlockPath(String path) throws IOException {
-    File localFolder;
     try {
-      localFolder = new File(PathUtils.getParent(path));
+      createStorageDirPath(PathUtils.getParent(path));
```

Comment:
That is intentional. 
The sticky bit is a permission bit that protects the files within a directory. If the directory has the sticky bit set, a file can be deleted only by the owner of the file, the owner of the directory, or by root. This special permission prevents a user from deleting other users' files from public directories such as `/tmp`. So, all caller of 'setLocalFileStickyBit()' surly first set the directory permission to "777", then
- when you set up a public directory, be sure to set the sticky bit
- For private directory, this setting will not influence owner's operations. For the directory is private, others users originally should not delete owner's file. 

So, this design is reasonable.
BTW, original method name `setLocalFileStickyBit` is somewhat misleading, `setLocalDirStickyBit` maybe a better choice. Any advice? 

Thanks for your review! @gpang